static-checks/nist-validation/test.py

```diff
@@ -27,12 +30,15 @@
         '-valresultfile', result_file,
         '-file', datastream,
     ]
-    proc = util.subprocess_run(cmd, stdout=subprocess.PIPE, check=True, universal_newlines=True)
-    if 'The target is valid' in proc.stdout:
+    util.subprocess_run(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, check=True)
+    tree = ET.parse(result_file)
+    root = tree.getroot()
+    failures = 0
+    for elem in root.findall('./nist:results/nist:base-requirement/[nist:status="FAIL"]', ns):
+        name = f'{ds_name}/{elem.attrib["id"]}'
+        results.report('fail', name, logs=[report_file, result_file])
+        failures += 1
+    if not failures:
         results.report('pass', ds_name)
```

Comment:
Let's search all requirements' statuses, report `fail` for `status="FAIL"`, `pass` for PASS, and `error` (?) for everything else. To have everything reported and to avoid the `failures` counted.

Reply:
@mildas Based on the scapval stdout where it summarizes results there are the following result types:
```
57 base requirements having 48 PASS, 6 WARNING, 1 FAIL, and 2 INFORMATIONAL results
```
Why do you want to report other results than FAIL? It will increase the amount of results in the report which I don't think we will be looking into.

Reply:
Okay, let me update it then.